lib/action_client/base.rb

```diff
@@ -3,34 +3,55 @@ class Base < AbstractController::Base
     include AbstractController::Rendering
     include ActionView::Layouts
 
-    cattr_accessor :defaults,
+    class_attribute :defaults,
       instance_accessor: true,
       default: ActiveSupport::OrderedOptions.new
 
+    class_attribute :middleware,
+      instance_accessor: true
+
+    class_attribute :after_actions,
+      instance_accessor: true,
+      default: []
+
     class << self
+      def inherited(descendant)
+        descendant.middleware = Rails.configuration.action_client.middleware.dup
+        descendant.after_actions = after_actions.dup
+      end
+
       def default(options)
         options.each do |key, value|
           defaults[key] = value
         end
       end
 
+      def after_action(&block)
+        after_actions.append(block)
+      end
+
+      def rescue_from(*rescued_status_codes, &block)
+        after_actions.append -> (status, headers, body) {
```

Comment:
It might be helpful to future readers if we re-use our public interface:

```suggestion
        after_action { |status, headers, body|
```

Reply:
Incorporated as part of https://github.com/thoughtbot/action_client/pull/3/commits/1f91c05cd543b452d049e31e01c1f50b400b4e11.